src/test/ui/consts/const-int-rotate.rs

```diff
@@ -3,4 +3,5 @@ fn main() {
     //~^ ERROR temporary value dropped while borrowed
     let y: &'static i32 = &(5_i32.rotate_right(3));
     //~^ ERROR temporary value dropped while borrowed
+    assert_eq!(0i16.rotate_left(124), 0);
```

Comment:
Uh, this is weird. @oli-obk what kind of test is this? I thought it would test functionality for rotate in CTFE but that does not seem to be the case. Do we have any test that does that?

@lzutao 	in any case, we want to test functionality here, so please move the actual call to within a `const` item. Something like
```rust
const X: i16 = 0i16.rotate_left(124);
```

Reply:
The above code just checks that we don't promote calls to the `rotate_*` methods. The actual tests are in https://github.com/rust-lang/rust/blob/9ebf47851a357faa4cd97f4b1dc7835f6376e639/src/test/run-pass/const-int-rotate.rs

Reply:
Ah, thanks!

Both this test here and that one suffer for a severe lack of comments (e.g. for stuff like the `ident` function).